### PR TITLE
Cxx: attach "end:" fields to struct variables 

### LINKED ISSUE #2498: c++: end line of a variable is not working for global struct variable in linux source
I find ctags supports end line of variable from https://github.com/universal-ctags/ctags/issues/2454.

I use this command to parse test.c.

## command:
`ctags.exe --output-format=json --langmap=c:+.inc --kinds-c=+plz --fields=+nie -o - d:\test.c`

## test.c
```
static const struct super_operations dax_sops = {
	.statfs = simple_statfs,
	.alloc_inode = dax_alloc_inode,
	.destroy_inode = dax_destroy_inode,
	.free_inode = dax_free_inode,
	.drop_inode = generic_delete_inode,
};

int n[2] = {
  1,
  2
};
```

## result
```
{"_type": "tag", "name": "dax_sops", "path": "D:\\tmp\\test\\al.c", "pattern": "/^static const struct super_operations dax_sops = {$/", "file": true, "line": 1, "typeref": "typename:const struct super_operations", "kind": "variable"}
{"_type": "tag", "name": "n", "path": "D:\\tmp\\test\\al.c", "pattern": "/^int n[2] = {$/", "line": 9, "typeref": "typename:int[2]", "kind": "variable", "end": 12}
```

The variable `dax_sops` does not have end line, while `n` has.

**PS**: `dax_sops` code snippet is copyed from super.c in linux source code (https://github.com/torvalds/linux/blob/master/drivers/dax/super.c#L482)

### PR DESCRIPTION
**Pull request by masatake**
Close #2498.

### CODE REVIEW

**Comment by pragmaware** [comment]
Sorry for the delay in reviewing. I have the children at home because of the COVID lockdown and it's very hard to find free time and brain resources to work on this. I'll be back more often once the situation settles a bit.

File: parsers/cxx/cxx_parser.c [hunk 1]
```diff
@@ -1105,6 +1106,10 @@ static bool cxxParserParseClassStructOrUnionInternal(
 			return false;
 		}
 
+		if (iCorkIndex != CORK_NIL)
+			cxxParserSetEndLineForTagInCorkQueue (iCorkIndex,
+												  cxxTokenChainLast(g_cxx.pTokenChain)->iLineNumber);
```
[the remark this answers is not in the document]
Comment (masatake) [reply parsers/cxx/cxx_parser.c:1111]: Thank you. I agree with you.

**Comment by masatake** [comment]
Thank you very much.
I wish you and your family are healthy.
Please, take your time for your family.
The goal of this project is extending the life of ctags.
For the goal, we, the developers of ctags must survive.

**Comment by masatake** [comment]
BTW, 4884407a999d5193b95ac8f4343294cd21b128c2 is still meaningful.
How do you think?

**Pull request closed without merging**